sync-core/src/main/java/com/cloudant/sync/datastore/AttachmentManager.java

```diff
@@ -159,7 +170,7 @@ protected PreparedAndSavedAttachments prepareAttachments(Collection<? extends At
         return preparedAndSavedAttachments;
     }
 
-    protected void setAttachments(SQLDatabase db,BasicDocumentRevision rev, PreparedAndSavedAttachments preparedAndSavedAttachments) throws IOException {
+    protected void setAttachments(SQLDatabase db,BasicDocumentRevision rev, PreparedAndSavedAttachments preparedAndSavedAttachments) throws AttachmentNotSavedException, AttachmentException {
```

Comment:
I'm not sure of the best pattern to use, I think it'll come down to the API design, where I think we should document all the exceptions that can be thrown.